flink-connectors/flink-orc/src/main/java/org/apache/flink/orc/RowOrcInputFormat.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.orc.RecordReader;
+import org.apache.orc.TypeDescription;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import static org.apache.flink.orc.OrcUtils.fillRows;
+
+/**
+ * InputFormat to read ORC data.
+ * For Optimization, reading is done in batch instead of a single row.
+ */
+public class RowOrcInputFormat
+	extends FileInputFormat<Row>
+	implements ResultTypeQueryable<Row> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(RowOrcInputFormat.class);
+	private static final int BATCH_SIZE = 1024;
+
+	private org.apache.hadoop.conf.Configuration config;
+	private TypeDescription schema;
+	private int[] fieldMapping;
+
+	private transient RowTypeInfo rowType;
+	private transient RecordReader orcRowsReader;
+	private transient VectorizedRowBatch rowBatch;
+	private transient Row[] rows;
+
+	private transient int rowInBatch;
+
+	public RowOrcInputFormat(String path, String schemaString, Configuration orcConfig) {
+		this(path, TypeDescription.fromString(schemaString), orcConfig);
+	}
+
+	public RowOrcInputFormat(String path, TypeDescription orcSchema, Configuration orcConfig) {
+		super(new Path(path));
+		this.unsplittable = false;
+		this.schema = orcSchema;
+		this.rowType = (RowTypeInfo) OrcUtils.schemaToTypeInfo(schema);
+		this.config = orcConfig;
+
+		this.fieldMapping = new int[this.schema.getChildren().size()];
+		for (int i = 0; i < fieldMapping.length; i++) {
+			this.fieldMapping[i] = i;
+		}
+
+	}
+
+	public void setFieldMapping(int[] fieldMapping) {
+		this.fieldMapping = fieldMapping;
+		// adapt result type
+
+		TypeInformation[] fieldTypes = new TypeInformation[fieldMapping.length];
+		String[] fieldNames = new String[fieldMapping.length];
+		for (int i = 0; i < fieldMapping.length; i++) {
+			fieldTypes[i] = this.rowType.getTypeAt(fieldMapping[i]);
+			fieldNames[i] = this.rowType.getFieldNames()[fieldMapping[i]];
+		}
+		this.rowType = new RowTypeInfo(fieldTypes, fieldNames);
+	}
+
+	private boolean[] computeProjectionMask() {
+		boolean[] projectionMask = new boolean[schema.getMaximumId() + 1];
+		for (int inIdx : fieldMapping) {
+			TypeDescription fieldSchema = schema.getChildren().get(inIdx);
+			for (int i = fieldSchema.getId(); i <= fieldSchema.getMaximumId(); i++) {
+				projectionMask[i] = true;
+			}
+		}
+		return projectionMask;
+	}
+
+	@Override
+	public void openInputFormat() throws IOException {
+		super.openInputFormat();
+		this.rows = new Row[BATCH_SIZE];
+		for (int i = 0; i < BATCH_SIZE; i++) {
+			rows[i] = new Row(fieldMapping.length);
+		}
+	}
+
+	@Override
+	public void open(FileInputSplit fileSplit) throws IOException {
+
+		this.currentSplit = fileSplit;
+		Preconditions.checkArgument(this.splitStart == 0, "ORC files must be read from the start.");
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Opening ORC file " + fileSplit.getPath());
```

Comment:
As you're using slf4j, you can write something like 
```
LOG.debug("Opening ORC file {}", fileSplit.getPath());
```

and avoid the if statement.